tests/test_publish.py

```diff
@@ -424,6 +424,8 @@ def test_publish(dbs, persistent_repository, version):
         number_of_media_files
         - number_of_media_archives
     )
+    # Check if media files are sorted
+    assert deps.media == sorted(deps.media)
```

Comment:
Good point, as we only sort `update_media` and `add_media` it will at newly sorted media at the end of the already existing ones. Also updated and added media are already pasted after each other I guess.

Reply:
I guess we can write a test for it and see

Reply:
So I guess if we really want to make sure it is sorted we need to do the sorting when storing the dependency CSV file.

But maybe it is also fine to only have the piecewise sorted stuff, at least we wanted to make sure that we don't have random order?

Reply:
I guess it's fine to keep it as is, since we only want to assure there is a deterministic order and since we by design publish media files in alphabetic order the test will not fail. I was just curious if my assumption is right. But maybe we should add a comment that in generally the files must not be ordered, but only in this specific test this is the case.

Reply:
To me it makes sense that we add new files in a certain order, but I don't really see a need for ordering the dependency table as a whole. The user can still order the table herself if she wants.

Reply:
So I did wrote an offline test to publish a first version with `z.wav` and then update it adding a file `a.wav`, and your assumption was correct, the returned value of the deps is `[z.wav, a.wav]`. 

But it makes sense to say that the current approach is fine, we're not interested in sorting but only in having the dependencies in a deterministic way. In this case I'll update the comments to communicate this as suggested

Reply:
I've added to the comments suggested by hagen in `publish.py` the last two lines
```python
    # Add updated and new media to dependencies
    # and sort them by paths
    # as they can be returned in random order
    # by the discovery job
    # Sorting enforces determinism in the
    # generation of the dependencies table
    if update_media:
        update_media = sorted(update_media, key=lambda x: x[0])
        deps._update_media(update_media)
    if add_media:
        add_media = sorted(add_media, key=lambda x: x[0])
        deps._add_media(add_media)
```

And also in `test_publish.py`
```python
    # Check if media files are sorted
    # This does mean that media files are
    # always sorted by alphabetical order
    # but only in this specific test case
    # Here we're testing for determinism rather
    # than ordering
    assert deps.media == sorted(deps.media)
```

Reply:
Great, thanks a lot.